ops/testing.py

```diff
@@ -0,0 +1,362 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from textwrap import dedent
+
+from ops import charm, framework, model
+
+
+def create_harness(charm_cls, charm_meta_yaml):
+    """Used for testing your Charm or component implementations.
+
+    This ensures that you have an instance of `charm_cls` that can be driven by a TestingHarness.
+
+    Example::
+
+        charm, harness = create_harness(MyCharm, '''
+            name: my-charm
+            requires:
+              db:
+                interface: pgsql
+            ''')
+        relation_id = harness.add_relation('db', 'postgresql')
+        harness.add_relation_unit(relation_id, 'postgresql/0', remote_unit_data={'key': 'value'})
+        # Check that charm has properly handled the relation_joined event for postgresql/0
+
+    :param charm_cls: The Charm class that should be tested. If you are just testing a component,
+        you can pass in ops.charm.CharmBase.
+    :rtype charm_cls: CharmBase
+    :param charm_meta_yaml: The YAML metadata for the charm, defining interfaces, name, etc.
+        This can be either a string or a file.
+    :return: (charm, harness)
+    :rtype: (CharmBase, TestingHarness)
```

Comment:
I thought we'd agreed to make this return just a harness, that had the charm as one of its attributes?

Reply:
This is one thing that got brought up, and the first time we talked about it, that is where we went. The second time we talked about it, I mentioned that it is 'one side is the object under test, and the other is the testing harness', and it sounded like that stuck.
That said, I have a follow up which uses the harness for test_model.py, and it is, indeed, true that we don't really ever use the charm return value there. (we make heavy use of charm.model to in the tests, but never the charm itself, so I'm ok pulling that out.)
I did slightly still prefer the clarity between what is fake and what isn't.

Shall I add a function like 'TestingHarnes.get_charm()' ?